common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java

```diff
@@ -610,8 +613,60 @@ public Bundle extractInteractiveRequestBundleFromResultBundle(@NonNull final Bun
         return resultBundle;
     }
 
+    /**
+     * Get authorizationResult from resultBundle for Device Code Flow
+     * @param resultBundle The bundle to interpret
+     * @return authorizationResult {@link AuthorizationResult}
+     * @throws BaseException
+     * @throws ClientException
+     */
+    @NonNull
+    public AuthorizationResult getDeviceCodeFlowAuthResultFromResultBundle(@NonNull final Bundle resultBundle) throws BaseException, ClientException {
+        String serializedDCFAuthResult = resultBundle.getString(AuthenticationConstants.Broker.BROKER_DCF_AUTH_RESULT);
+        if (serializedDCFAuthResult != null) {
+            AuthorizationResult authorizationResult = ObjectMapper.deserializeJsonStringToObject(serializedDCFAuthResult, MicrosoftStsAuthorizationResult.class);
+            return authorizationResult;
+        }
+
+        // DCF not supported - thrown when BrokerFlight.ENABLE_DCF_IN_BROKER is false
+        BrokerResult brokerResult = brokerResultFromBundle(resultBundle);
+        if (brokerResult.getErrorCode() != null && brokerResult.getErrorCode().equals(ErrorStrings.DEVICE_CODE_FLOW_NOT_SUPPORTED)) {
+            Logger.error(TAG, "deviceCodeFlowAuthRequest() not supported in BrokerMsalController", new ClientException(ErrorStrings.DEVICE_CODE_FLOW_NOT_SUPPORTED, "deviceCodeFlowAuthRequest() not supported in BrokerMsalController"));
+            throw new ClientException(ErrorStrings.DEVICE_CODE_FLOW_NOT_SUPPORTED, "deviceCodeFlowAuthRequest() not supported in BrokerMsalController");
+        }
+
+        throw getBaseExceptionFromBundle(resultBundle);
+    }
+
+    /**
+     * Get acquireTokenResult from resultBundle for Device Code Flow
+     * @param resultBundle The bundle to interpret
+     * @return acquireTokenResult {@link AcquireTokenResult}
+     * @throws BaseException
+     * @throws ClientException
+     */
     @NonNull
```

Comment:
Here we have nonNull return. But we are returning null at line 662. Please change this accordingly

Reply:
done